apps/exam/views.py

```diff
@@ -46,33 +46,55 @@ def get_context_data(self, **kwargs):
     """Render intro page template for chosen exam. """
 
     context = super().get_context_data(**kwargs)
-    exam = Exam.objects.get(id=self.kwargs['exam_id'])
-    context['exam'] = exam
-    context[
-        'attempt_duration'] = exam.duration_minutes + self.request.user.required_extra_time
+    if 'exam_id' in self.kwargs:
+      exam = Exam.objects.filter(id=self.kwargs['exam_id'])
+      context['subject'] = Subject.objects.get(exam=exam[0])
+      context['exam'] = exam[0]
+
+    if 'subject_id' in self.kwargs:
+      context['subject'] = Subject.objects.get(id=self.kwargs['subject_id'])
+      exam = Exam.objects.filter(subject_id=self.kwargs['subject_id'])
```

Comment:
Here you have more that one exam -> exams

Reply:
Yes, but I need the same name for both cases (when it's subject attempt and a regular exam) because I use it  to calculate attempt_duration later on in this method. If I rename it - it won't be DRY compliant.